services/src/main/java/org/openlearn/service/StorageService.java

```
@@ -109,23 +144,101 @@ public InputStream getUpload(Long assignmentId, Long portfolioId, String fileNam
 		}
 	}
 
-	public void deleteUpload(Long assignmentId, Long portfolioId, String fileName) {
-		AmazonS3 s3client = new AmazonS3Client(new ProfileCredentialsProvider());
-		String uploadBucketName = props.getUploadBucket();
+	public void deleteUpload(Long fileInformationId) {
+		FileInformation fileInformation = fileRepository.findOne(fileInformationId);
+
+		if (fileInformation == null) throw new FileInformationNotFoundException(fileInformationId);
+		deleteUpload(fileInformation);
+	}
+
+	public void deleteUpload(FileInformation fileInformation) {
+		AmazonS3 s3client = AmazonS3ClientBuilder.defaultClient();
+		String bucket = retrieveBucket(fileInformation);
+		String key = retrieveKeyName(fileInformation);
+
 		try {
-			String key = getS3FilePrefix(assignmentId, portfolioId) + fileName;
-			s3client.deleteObject(new DeleteObjectRequest(uploadBucketName, key));
+			s3client.deleteObject(new DeleteObjectRequest(bucket, key));
+			fileRepository.delete(fileInformation);
 		} catch(Exception e) {
 			log.error(e.getMessage());
```

This exception is getting swallowed.

---

Yeah, holdover from the first implementation, it's in the works of being fixed.